dc/s2n-quic-dc/src/stream/server/tokio/tcp/worker.rs

```diff
@@ -226,25 +258,130 @@ enum WorkerState {
 }
 
 impl WorkerState {
-    fn poll<Sub, Pub>(
+    fn poll<Sub, Pub, B>(
         &mut self,
         cx: &mut task::Context,
-        context: &mut Context<Sub>,
+        context: &mut Context<Sub, B>,
         stream: &mut Option<(LazyBoundStream, SocketAddress)>,
         subscriber_ctx: &mut Option<Sub::ConnectionContext>,
         queue_time: Timestamp,
         now: Timestamp,
         publisher: &Pub,
+        unix_socket_path: &Option<String>,
     ) -> Poll<Result<ControlFlow<()>, Option<io::Error>>>
     where
         Sub: event::Subscriber + Clone,
         Pub: EndpointPublisher,
+        B: PollBehavior<Sub>,
+    {
+        B::poll(
+            self,
+            cx,
+            context,
+            stream,
+            subscriber_ctx,
+            queue_time,
+            now,
+            publisher,
+            unix_socket_path,
+        )
+    }
+
+    #[inline]
+    fn poll_initial_packet<Pub>(
```

Comment:
I think this is still `impl WorkerState`, why did you move it up here? Generally speaking, prefer to keep code movement to separate commits / PRs to make it clearer when there's _edits_ vs. just moving around code.

Reply:
I don't think I actually moved `poll_initial_packet`. I moved the `poll` implementation to the trait below but the diff shows `poll_initial_packet` as being moved...